### PR TITLE
fixing up redemptions container

### PR DESCRIPTION
**Pull request by jellegerbrandy**
redemptions container had a large numer of bugs

FIXED in this PR:

- write `+0.00%` instead of `0.00%` if the value is not zero 
- `+0.00%` is still super uninformative, so I added the complate absolute value in the tooltip
- remove the isredeempending checks which are not working in any cases
- also take into account cases in which rewards were already distributed
- show only proposals that are actually executable (work around an upstream bug which is here https://github.com/daostack/subgraph/issues/252)
- #731 [cannot redeem because proposals already redemed, fixed by line above]
- #712 [wrong totals]  - is the same problem as 731, I think
- show all different GP rewards in redemptions tip
- fix logic of "you vs the beneficiary will receive" in redemptions tip
- remove the redemptionsstring (next to the redeem button) as it is was showing the wrong results and it is unclear what is should show and it would take me 2 hours to fix the convoluted logic
- active redeem button on non-contributionreward proposals (where is was not shown at all)
- fix logic of showing redeem button on proposaldetaisl container
- **fix bug where we were actually never claiming our own rewards, but only the ones for the beneficiary**
- #588 [reward display not updated] [Seems fixed now, cannot reproduce]
- the "redeem for beneificiary" button was showing also on failed proposals. I added ane xtra check, so it does not, but the underlying bug is https://github.com/daostack/subgraph/issues/263
- factor out logic of contributionreward's claimable rewards, and fix some of it

RELATED:
- #760 [write tests]
- #658 [redemption tip]
- #732 (needs input from @A-Zak )
 #703 [redeem your stakes on timed-out proposals] 
- #643 [duplicate of 703]
- #758 [try to distribute from invalid token address, needs input on how to handle it]

### CODE REVIEW

File: src/components/ViewDao/DaoRedemptionsContainer.tsx [hunk 1]
```diff
@@ -110,22 +114,29 @@ export default (props: { dao: IDAOState, currentAccountAddress: Address } & Rout
     proposals(where: {
       accountsWithUnclaimedRewards_contains: ["${props.currentAccountAddress}"]
       dao: "${props.dao.address}"
-      contributionReward_not: null
+      stage_in: ["Boosted", "QuietEndingPeriod", "Executed"]
```
Comment (tibetsprague) [inline src/components/ViewDao/DaoRedemptionsContainer.tsx]: why would we want boosted proposals here?

Comment (jellegerbrandy) [reply src/components/ViewDao/DaoRedemptionsContainer.tsx]: oops, good point.

Just the "executed" proposals, then

Comment (jellegerbrandy) [reply src/components/ViewDao/DaoRedemptionsContainer.tsx]: when we upgrade the the new version of the subgraph, this line will be unecessary : https://github.com/daostack/subgraph/issues/252

**Comment by A-Zak** [comment]
A few regression we have with all these changes...
• When creating a proposal I get this error in console:
image

I'm guessing that fixing this will also fix:
• an error displayed on proposal card
image

• Crash on redemptions page (with potential rewards)
address: 0x6d1A4be65723355a9f5b12C446780a5000849461
https://daostack-alchemy-stagin-pr-778.herokuapp.com/dao/0x3cdb39c1ca57c19a316dd2f5acf900705220c713/redemptions/
image

**Comment by jellegerbrandy** [comment]
* fixed the crashing redemptions page

**Comment by jellegerbrandy** [comment]
Ok, I believe the first (2) bugs were fixed by @tibetsprague 's commit, I just fixed the last one (crashing redemptions page)

File: src/components/Proposal/ActionButton.tsx [hunk 2]
```diff
@@ -1,39 +1,37 @@
...
 import { showNotification } from "reducers/notifications";
 import { IProfileState } from "reducers/profilesReducer";
 import { Observable, of} from "rxjs";
-import { isRedeemPending } from "selectors/operations";
-import RedemptionsString from "./RedemptionsString";
-import RedemptionsTip from "./RedemptionsTip";
-
+import { map, mergeMap } from "rxjs/operators";
 import * as css from "./ActionButton.scss";
+/* import RedemptionsString from "./RedemptionsString"; */
```
Comment (tibetsprague) [inline src/components/Proposal/ActionButton.tsx]: remove commented out stuff

File: src/components/Proposal/ActionButton.tsx [hunk 3]
```diff
@@ -144,43 +139,43 @@ class ActionButton extends React.Component<IProps, IState> {
...
             <button className={css.executeButton} onClick={this.handleClickExecute.bind(this)}>
               <img src="/assets/images/Icon/execute.svg"/>
               { /* space after <span> is there on purpose */ }
               <span> Execute</span>
             </button>
           : redeemable ?
             <div>
-              { !detailView ?
-                  <RedemptionsString currentAccountAddress={currentAccountAddress} dao={dao} proposal={proposal} rewards={rewardsForCurrentUser} />
+              {/* !detailView ?
```
Comment (tibetsprague) [inline src/components/Proposal/ActionButton.tsx]: we dont want this? or its just broken for now?

File: src/components/Proposal/Predictions/PredictionButtons.tsx [hunk 4]
```diff
@@ -258,7 +257,7 @@ class PredictionBox extends React.Component<IProps, IState> {
               }
             </span>
             : <span className={css.disabledPredictions}>
-               Predictions are disabled
+                { /* Predictions are disabled */ }
```
Comment (tibetsprague) [inline src/components/Proposal/Predictions/PredictionButtons.tsx]: ?

File: src/components/ViewDao/DaoRedemptionsContainer.tsx [hunk 5]
```diff
@@ -36,34 +36,42 @@ class DaoRedemptionsContainer extends React.Component<IProps, null> {
     // TODO: the reward object from the subgraph only gives rewards for voting and staking and dao bounty,
     // the original code also considers ethREward and externalTokenRewards
     // let ethReward = 0
+
+    // TODO: move all the next logic to the client library
+    // calculate the total rewards from the genesisprotocol
     const genReward = new BN("0");
+    const ethReward = new BN("0");
     const reputationReward = new BN(0);
     // , externalTokenReward = 0;
```
Comment (tibetsprague) [inline src/components/ViewDao/DaoRedemptionsContainer.tsx]: shouldnt we add in external token rewards now too?

File: src/lib/util.ts [hunk 6]
```diff
@@ -199,3 +199,35 @@ export function linkToEtherScan(address: Address) {
...
   return `https://${prefix}etherscan.io/address/${address}`;
 }
+
+export function getClaimableRewards(reward: IRewardState) {
+  if (!reward) {
+    return {};
+  }
+
+  const result: {[key: string]: BN} = {};
+  if (reward.reputationForProposer.gt(new BN(0)) && new BN(reward.reputationForProposerRedeemedAt).isZero()) {
```
Comment (tibetsprague) [inline src/lib/util.ts]: why are you converting the RedeemedAt fields to BigNumbers? arent they dates?

File: src/components/Proposal/ActionButton.tsx (hunk 3 above)
Comment (jellegerbrandy) [reply src/components/Proposal/ActionButton.tsx]: quoting from the PR description: "remove the redemptionsstring (next to the redeem button) as it is was showing the wrong results and it is unclear what is should show"

I thiknk we are making a mistake showing similar info in different formats in different places in slightly different ways, and we should rethink that approach.

File: src/components/Proposal/Predictions/PredictionButtons.tsx (hunk 4 above)
Comment (jellegerbrandy) [reply src/components/Proposal/Predictions/PredictionButtons.tsx]: that text is superconfusing to me, every time I see it. I created an issue and will put it back for now

File: src/components/ViewDao/DaoRedemptionsContainer.tsx (hunk 5 above)
Comment (jellegerbrandy) [reply src/components/ViewDao/DaoRedemptionsContainer.tsx]: perhaps, I don't know, there could be 100 different tokens, so it's a whole new ballgame.

File: src/lib/util.ts (hunk 6 above)
Comment (jellegerbrandy) [reply src/lib/util.ts]: because of a client bug. Now fixed.

**Pull request merged**